src/screens/Manager/Manager.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useCallback, useEffect } from "react";
+import { NavigationActions } from "react-navigation";
+import type { Action, State } from "@ledgerhq/live-common/lib/apps";
+
+import { ManagerContext } from "./shared";
+import AppsScreen from "./AppsScreen";
+import GenericErrorBottomModal from "../../components/GenericErrorBottomModal";
+import QuitManagerModal from "./Modals/QuitManagerModal";
+import StorageWarningModal from "./Modals/StorageWarningModal";
+import AppDependenciesModal from "./Modals/AppDependenciesModal";
+import UninstallDependenciesModal from "./Modals/UninstallDependenciesModal";
+
+type Props = {
+  screenProps: {
+    state: State,
+    dispatch: Action => void,
+  },
+  navigation: *,
+};
+
+export default ({ screenProps: { state, dispatch }, navigation }: Props) => {
+  const { apps, currentError, installQueue, uninstallQueue } = state;
+  const [quitManagerAction, setQuitManagerAction] = useState(false);
+
+  useEffect(() => {
+    const blockNavigation = installQueue.length + uninstallQueue.length > 0;
+    const n = navigation.dangerouslyGetParent();
+    if (n) {
+      n.setParams({ blockNavigation });
+      let navListener;
+      if (blockNavigation) {
+        navListener = navigation.addListener("action", e => {
+          if (e.action && e.action.type !== NavigationActions.SET_PARAMS) {
+            setQuitManagerAction(e.action);
+          }
+        });
+      } else if (navListener) {
+        navListener.remove();
+        setQuitManagerAction(false);
+      }
+    }
+  }, [installQueue.length, uninstallQueue.length, setQuitManagerAction]);
+
+  const [error, setError] = useState(null);
+  useEffect(() => setError(currentError), [setError, currentError]);
```

Comment:
can you move all the useState at the beginning to make it a bit more clear maybe?

I also suggest to prepend a comment to explain what this is doing? if i'm correct: `// open error modal each time a new error appears in state.currentError`